### ISSUE TITLE
Trains Entering Manual Mode

### ISSUE THREAD

**Issue by RocketMan1081** [LuziferSenpai/Automatic-Coupling-System#8]
When uncoupling wagons, the trains do not return to automatic mode and remain in manual mode.  I do have a heavily modded game, but I was able to isolate this with just this and the inventory sensor mods installed.  

Let me know if you want screenshots, files, etc

**Comment by LuziferSenpai** [comment]
Are you on the latest version?
How does your setup look? (A GIF would be best I think)
Does the Tutorial in Tips & Tricks work correctly?

**Comment by RocketMan1081** [comment]
Hi, sorry for the delayed response.
I am running the latest version.  I do have a very heavily modded save going.  I can provide a video, a GIF, and even the blueprint of my setup (it is a trial loop to test and understand the mod before I integrate it into the base.  I do not see a tutorial in the tips and tricks window.

<iframe src="https://giphy.com/embed/DkgMU4YtQss1TIXUOL" width="480" height="270" frameBorder="0" class="giphy-embed" allowFullScreen></iframe><p><a href="https://giphy.com/stickers/transparent-DkgMU4YtQss1TIXUOL">via GIPHY</a></p>

**Comment by RocketMan1081** [comment]
It seems the GIFs are not working.  Here is a link to this video on my YouTube channel.  [Link](https://youtu.be/83cptF6ySmM)

**Comment by AdamBaston** [comment]
Hi, I'm getting the same issue. I've isolated it to only happen when there are wagons attached to the outside of the locomotives after uncoupling.

The Tips and Tricks works as expected as it does not leave any wagons attached to the outside of locomotives.

You can reproduce this by setting a stations' uncouple signal to 1 instead of 2 in the tutorial.

**Comment by RocketMan1081** [comment]
Ok, I understand.  What I am trying to do in the video is that I am coupling to the 16 wagons left by other locomotives, then move slightly to the left and uncouple all but 4 wagons to take away (leaving 12 behind).

The stations each have a signal "coupling" >0 in which this is 16, and the second is "uncoupling" >0 in which this is 6 [to keep the two locomotives and 4 wagons].

**Comment by LuziferSenpai** [comment]
Yea, i will check to fix this tomorrow. Sry that it took so long to answer, I'm a bit busy with work

**Comment by LuziferSenpai** [comment]
Ups, deleted the wrong comment.
SRY!

But this should be fixed with 2.0.3, pls give me feedback

**Comment by AdamBaston** [comment]
Thanks for pushing an update, it has fixed manual mode when changing direction after uncoupling (like in the Tips and Tricks) but going through a station (and not changing direction) will still enable manual mode, regardless if a wagon is left behind or not.

See below for a blueprint, changing the combinator to 2 instead of 3 causes the train to still remain in manual mode

<details>
  <summary>Blueprint</summary>
  
`0eNqlV8GymjAU/ZVO1uCQBEXctf2EtquOw0RIMdOQMCH46jj8exO06hsvi/u6UZHck3A495zkQg56lL1TxpPdhajamoHsfl7IoFojdPzPn3tJdkR52ZGEGNHFKyeUJlNClGnkH7Kj0z4h0njllbzWzxfnyozdQbow4F45+FDbHn06QySkt0OosiZOFZBSuk7IOXzn62lKXnAYHodDOPyOE2FMOnjbQyD8DpKEGcX1HvlCAMgcvzQKLW2NxuElhLPB4xQQToHHAV/dFo8DvroSjwPyTDM0EAOJpnhxM5Bpilc3A6mmHA8Eck3xomYw2XhVU5hsvKwpTDZe1xQmGy9sCpONVzYFyWYZzt0CSkKC83tndXWQR3FS1s3WH6urWN3LpnqJhJNyfhT6kQrXEel3Ml0BjazjfEOsofGjdVKa53xQTVgum/bTO3/9DPkrw7cZqCD26LKYdl4Yn9a2OygjfHhsAI2u/uGt1gtM/VLaS7cQnUs8NbK2Y68lmUHHGMH8HqmRFAyJWSQRely8F4D9wvBWACf4B/INxPmAEYA4Dx/QNujAenWSyyBsVZabrORFVIJ1KqDdVJtB4GhvgNeIdgZ404NOPPANcopirFxmLLbTKzw6BUG9crTuQbfgaNnDnszRsodDgj9kXwvX2vRNtKF0GWWzTH/0mlO4ti4MNKPW0HzomIRTkuM7AX6t6FaA9xF5huOR8f/jMUenF7yRytHdAe/scnR7wFvNHN0fce8bTotDfZTNqG/HxYeRxOvgyZw+jbnG6tP2ICFvQvkqpGOjbuEYRgSIXjhZ3WJ3DvPbb6+6aFFe1b/D2E0My+Ty/kS3j6uaT7m7p0NxQrQ4SB1n/frt0w8zJ7YybbhxCoE/V7MtzYuSFdsto7zYTNNfrSAFzg==`
  
</details>

**Comment by LuziferSenpai** [comment]
> Thanks for pushing an update, it has fixed manual mode when changing direction after uncoupling (like in the Tips and Tricks) but going through a station (and not changing direction) will still enable manual mode, regardless if a wagon is left behind or not.
> 
> See below for a blueprint, changing the combinator to 2 instead of 3 causes the train to still remain in manual mode
> 
> Blueprint
> `0eNqlV8GymjAU/ZVO1uCQBEXctf2EtquOw0RIMdOQMCH46jj8exO06hsvi/u6UZHck3A495zkQg56lL1TxpPdhajamoHsfl7IoFojdPzPn3tJdkR52ZGEGNHFKyeUJlNClGnkH7Kj0z4h0njllbzWzxfnyozdQbow4F45+FDbHn06QySkt0OosiZOFZBSuk7IOXzn62lKXnAYHodDOPyOE2FMOnjbQyD8DpKEGcX1HvlCAMgcvzQKLW2NxuElhLPB4xQQToHHAV/dFo8DvroSjwPyTDM0EAOJpnhxM5Bpilc3A6mmHA8Eck3xomYw2XhVU5hsvKwpTDZe1xQmGy9sCpONVzYFyWYZzt0CSkKC83tndXWQR3FS1s3WH6urWN3LpnqJhJNyfhT6kQrXEel3Ml0BjazjfEOsofGjdVKa53xQTVgum/bTO3/9DPkrw7cZqCD26LKYdl4Yn9a2OygjfHhsAI2u/uGt1gtM/VLaS7cQnUs8NbK2Y68lmUHHGMH8HqmRFAyJWSQRely8F4D9wvBWACf4B/INxPmAEYA4Dx/QNujAenWSyyBsVZabrORFVIJ1KqDdVJtB4GhvgNeIdgZ404NOPPANcopirFxmLLbTKzw6BUG9crTuQbfgaNnDnszRsodDgj9kXwvX2vRNtKF0GWWzTH/0mlO4ti4MNKPW0HzomIRTkuM7AX6t6FaA9xF5huOR8f/jMUenF7yRytHdAe/scnR7wFvNHN0fce8bTotDfZTNqG/HxYeRxOvgyZw+jbnG6tP2ICFvQvkqpGOjbuEYRgSIXjhZ3WJ3DvPbb6+6aFFe1b/D2E0My+Ty/kS3j6uaT7m7p0NxQrQ4SB1n/frt0w8zJ7YybbhxCoE/V7MtzYuSFdsto7zYTNNfrSAFzg==`

Well ... That is something else, somehow the logic for decoupling doesnt go in the right direction.
For decoupling in front of the station you would use a negative number, not a positive.
Will need to check out now why the hell this doesnt work correctly.

I really need to check the logic complety now, because I'm confused myself.

**Comment by LuziferSenpai** [comment]
Okay, this is fixed for now with 2.0.4

**Comment by RocketMan1081** [comment]
Yes, it was something I had happen for moving in the same direction, my workaround was to let the locomotives sit for one second at the station before leaving, that seemed to give it enough time to register what I wanted it to do, I now have a seamless operation with your 2.0.3 release, have not updated to 2.0.4 yet.

**Comment by AdamBaston** [comment]
2.0.4 seems to have fixed my issue, thanks for fixing it so quickly!
